src/lib/core/fiber.c

```diff
@@ -1030,14 +1052,22 @@ fiber_loop(MAYBE_UNUSED void *data)
 		fiber_on_stop(fiber);
 		/* reset pending wakeups */
 		rlist_del(&fiber->state);
+		bool is_reusable = (fiber->flags & FIBER_CUSTOM_STACK) == 0;
```

Comment:
You have two checks if a fiber is reusable - here and in `fiber_join`. Let's add an inline helper function `fiber_is_reusable` with a proper comment so that we don't have to patch them both if we need to do some other kind of non-reusable fibers. The `is_reusable` argument of `fiber_recycle` looks redundant - we could use the helper function there as well.

Reply:
Done.